htdocs/contact/class/contact.class.php

```diff
@@ -1684,48 +1684,51 @@ public function updateRoles()
 
 		$error = 0;
 
-		$this->db->begin();
-
-		$sql = "DELETE FROM ".MAIN_DB_PREFIX."societe_contacts WHERE fk_soc=".$this->socid." AND fk_socpeople=".$this->id; ;
+		if ($this->socid > 0 || $this->socid == -1) {
+			$this->db->begin();
 
-		dol_syslog(__METHOD__, LOG_DEBUG);
-		$result = $this->db->query($sql);
-		if (!$result) {
-			$this->errors[] = $this->db->lasterror().' sql='.$sql;
-			$error++;
-		} else {
-			if (count($this->roles) > 0) {
-				foreach ($this->roles as $keyRoles => $valRoles) {
-					$sql = "INSERT INTO ".MAIN_DB_PREFIX."societe_contacts";
-					$sql .= " (entity,";
-					$sql .= "date_creation,";
-					$sql .= "fk_soc,";
-					$sql .= "fk_c_type_contact,";
-					$sql .= "fk_socpeople) ";
-					$sql .= " VALUES (".$conf->entity.",";
-					$sql .= "'".$this->db->idate(dol_now())."',";
-					$sql .= $this->socid.", ";
-					$sql .= $valRoles." , ";
-					$sql .= $this->id;
-					$sql .= ")";
-					dol_syslog(__METHOD__, LOG_DEBUG);
+			$sql = "DELETE FROM " . MAIN_DB_PREFIX . "societe_contacts WHERE fk_socpeople=" . $this->id;;
```

Comment:
The  AND fk_soc=".$this->socid is removed ? It means that if contact is default contact also for company B, when you update the value for company C, you will delete all links with company B ?

Reply:
Yes, I fixed it and only remove old default contact links if company was modified (new one or unlinked)